src/ripple/overlay/impl/OverlayImpl.cpp

```diff
@@ -604,6 +608,34 @@ OverlayImpl::size()
     return m_publicKeyMap.size ();
 }
 
+Json::Value
+OverlayImpl::crawl()
+{
+    Json::Value jv;
+    auto& av = jv["active"] = Json::Value(Json::arrayValue);
+    std::lock_guard <decltype(mutex_)> lock (mutex_);
+    for (auto const& e : m_publicKeyMap)
+    {
+        auto const sp = e.second.lock();
+        if (sp)
+        {
+            auto& pv = av.append(Json::Value(Json::objectValue));
+            pv["type"] = "peer";
+            pv["public_key"] = beast::base64_encode(
+                sp->getNodePublic().getNodePublic().data(),
+                    sp->getNodePublic().getNodePublic().size());
+            if (sp->crawl())
+            {
+                if (sp->slot()->inbound())
+                    pv["ip"] = sp->getRemoteAddress().address().to_string();
```

Comment:
Huh.  Just for thought, when I'm under a lock I try to get in and get out.  I try to collect information into a local light weight object, exit the lock, and then do any heavy-weight editing.  Does it make sense to be editing the Json::objectValue under the lock?  Maybe it's really light weight, so I shouldn't worry...